x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/action/TransportPreviewTransformAction.java

```diff
@@ -101,120 +129,122 @@ protected void doExecute(Task task,
         ClusterState clusterState = clusterService.state();
 
         final TransformConfig config = request.getConfig();
-        for(String src : config.getSource().getIndex()) {
-            String[] concreteNames = indexNameExpressionResolver.concreteIndexNames(clusterState, IndicesOptions.lenientExpandOpen(), src);
-            if (concreteNames.length == 0) {
-                listener.onFailure(new ElasticsearchStatusException(
-                    TransformMessages.getMessage(TransformMessages.REST_PUT_TRANSFORM_SOURCE_INDEX_MISSING, src),
-                    RestStatus.BAD_REQUEST));
-                return;
-            }
+        ValidationException validationResult = SourceDestValidator.validateForPreview(
+            clusterState,
+            indexNameExpressionResolver,
+            this.transportService.getRemoteClusterService(),
+            isRemoteSearchEnabled ? new RemoteClusterLicenseChecker(client, XPackLicenseState::isTransformAllowedForOperationMode) : null,
+            config.getSource().getIndex(),
+            config.getDestination().getIndex(),
+            clusterService.getNodeName(),
+            "basic"
```

Comment:
```suggestion
            License.OperationMode.BASIC.description()
```
We should use the operation mode enum.